alacritty_terminal/src/grid/resize.rs

```diff
@@ -144,6 +147,16 @@ impl<T: GridCell + Default + PartialEq + Copy> Grid<T> {
                 } else {
                     // Since we removed a line, rotate down the viewport.
                     self.display_offset = self.display_offset.saturating_sub(1);
+
+                    // Rotate cursors down if content below them was pulled from history.
+                    if i < cursor_buffer_line {
+                        self.cursor.point.line += 1;
+                    }
+
+                    let saved_buffer_line = (self.lines - self.saved_cursor.point.line - 1).0;
+                    if i < saved_buffer_line {
+                        self.saved_cursor.point.line += 1;
+                    }
```

Comment:
I'm not sure how important this is, since we usually don't expect content to be below the cursor during resize.

However since this does delete things *above* the cursor and not below it, I do think it makes some sense at least for `self.cursor`.

Reply:
The content below the cursor doesn't matter and can be safely cleared most of the time (maybe not all the time though). Alt screen is completely different here, but we don't reflow it, so fine.